pkg/autoscaler/stats_scraper.go

```diff
@@ -132,22 +143,31 @@ func (s *ServiceScraper) Scrape() (*StatMessage, error) {
 
 	sampleSize := populationMeanSampleSize(readyPodsCount)
 	statCh := make(chan *Stat, sampleSize)
+	scrapedPods := &safeStringSet{set: sets.NewString()}
 
 	grp := errgroup.Group{}
 	for i := 0; i < sampleSize; i++ {
 		grp.Go(func() error {
-			stat, err := s.sClient.Scrape(s.url)
-			if err != nil {
-				return err
+			for tries := 0; ; tries++ {
```

Comment:
https://play.golang.org/p/HH3bb9qU4x0 -- reflows the code with
- loops starting at 1, then check is against maxretries (no -1)
- code is indented one level less
- continue is gone

Reply:
Great suggestions! Adopted.